src/drt/src/dr/FlexGridGraph.h

```diff
@@ -755,61 +748,39 @@ class FlexGridGraph
   // must be safe access because idx1 and idx2 may be invalid
   void setGuide(frMIdx x1, frMIdx y1, frMIdx x2, frMIdx y2, frMIdx z)
   {
-    // if (!(isValid(x1, y1, z) && isValid(x2, y2, z))) {
-    //   return;
-    // }
     if (x2 < x1 || y2 < y1) {
       return;
     }
-    if (getZDir(z)) {  // H
+    if (getZDir(z) == Horizontal) {
```

Comment:
Since the routing direction is well defined it might be good to switch these to switch statements so that the compiler can warn us if there's ever a new enum type that we need to handle.

That goes for all instances of these checks

Reply:
I think that might be good for readability, but currently `getZDir` is reading from a `vector<bool>` which caches the layer directions. If there were an item added to the Enum, the whole data structure would have to change.

I'm debating about whether the whole layer caching thing should be removed because it doesn't save that much space, maybe O(# of layers) bytes.

Reply:
As of a few months ago this function was .143% of the profile. 

![image](https://user-images.githubusercontent.com/3784748/132908127-84636707-a569-437a-bc76-638d81c03865.png)

I would say remove the caching. That would reduce the complexity a lot, and would make contributing easier.

Reply:
Sounds good - I'm just kind of going down a rabbit hole of lots of things I want to fix with DRT. I'll add that to the list.

Reply:
Yeah very cool work though! Nice job @rovinski